src/nvim/shada.c

```diff
@@ -1800,6 +1803,14 @@ static ShaDaWriteResult shada_pack_entry(msgpack_packer *const packer,
         PACK_STATIC_STR(REG_KEY_WIDTH);
         msgpack_pack_uint64(spacker, (uint64_t) entry.data.reg.width);
       }
+      if (!CHECK_DEFAULT(entry, reg.is_unnamed)) {
```

Comment:
You have not specified the default. Currently it works only because when using `{ .attr = value }` syntax if some `.attr` is missing it is initialized with zeroes. Search for `DEF_SDE(Register` above.